actors/builtin/miner/miner_test.go

```diff
@@ -728,6 +732,79 @@ func TestWindowPost(t *testing.T) {
 			actor.submitWindowPoSt(rt, deadline, partitions, infos, cfg)
 		})
 	})
+
+	t.Run("detects faults from previous missed posts", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		// skip two PoSts
+		_, infos1, _ := runTillFirstDeadline(rt)
+		_, infos2, _ := runTillNextDeadline(rt)
+		deadline, infos3, partitions := runTillNextDeadline(rt)
+
+		// expect power to be deducted for all sectors in first two deadlines
+		rawPower, qaPower := miner.PowerForSectors(actor.sectorSize, append(infos1, infos2...))
+
+		// expected penalty is the undeclared fault penalty for all sectors in previous deadline
+		// and the undeclared late penalty for new faults preceding that.
+		_, qaPower1 := miner.PowerForSectors(actor.sectorSize, infos1)
+		expectedPenalty1 := miner.PledgePenaltyForLateUndeclaredFault(actor.epochReward, actor.networkQAPower, qaPower1)
+		_, qaPower2 := miner.PowerForSectors(actor.sectorSize, infos2)
+		expectedPenalty2 := miner.PledgePenaltyForUndeclaredFault(actor.epochReward, actor.networkQAPower, qaPower2)
+
+		cfg := &poStConfig{
+			skipped:               bitfield.NewFromSet(nil),
+			expectedRawPowerDelta: rawPower.Neg(),
+			expectedQAPowerDelta:  qaPower.Neg(),
+			expectedPenalty:       big.Add(expectedPenalty1, expectedPenalty2),
+		}
+
+		actor.submitWindowPoSt(rt, deadline, partitions, infos3, cfg)
+
+		// adds sectors as faults
+		st := getState(rt)
+		faultCount, err := st.Faults.Count()
```

Comment:
nit: seems about as much work and a stricter condition to check that the bitfield is set at the right sector numbers. I might be missing something though since this test is new to me.

Reply:
Now doing both.